File: 075/cpp/main.cpp
class Solution {
public:
    void swap(vector<int>& nums, int i, int j) {
        int n = nums[i];
        nums[i] = nums[j];
        nums[j] = n;
    }

    void sortColors(vector<int>& nums) {
        int i = 0;
        int j = nums.size() - 1;
        int k = 0;
        while (i <= k && k <= j) {
            while (i <= k && k <= j) {
                int n = nums[k];
                if (n == 0) {
                    swap(nums, i, k);
                    i += 1;
                } else if (n == 2) {
                    swap(nums, k, j);
                    j -= 1;
                } else {
                    break;
                }
            }
            k += 1;
        }
    }
};

File: 075/rust/main.rs
impl Solution {
    pub fn sort_colors(nums: &mut Vec<i32>) {
        let mut i = 0;
        let mut j = nums.len() - 1;
        let mut k = 0;
        while i <= k && k <= j {
            while i <= k && k <= j {
                let n = nums[k];
                if n == 0 {
                    nums.swap(i, k);
                    i += 1;
                } else if n == 2 {
                    nums.swap(k, j);
                    if j == 0 {
                        return;
                    }
                    j -= 1;
                } else {
                    break;
                }
            }
            k += 1;
        }
    }
}
